### ISSUE TITLE
"explicit parent script" not working

### ISSUE THREAD

**Issue by boneskull** [testdouble/scripty#20]
Contents of `scripts`:

```
scripts/
├── bump
│   ├── index.sh
│   ├── major.sh
│   ├── minor.sh
│   └── patch.sh
├── preversion.sh
├── publish-please.sh
├── test
│   ├── browser
│   │   ├── dev.sh
│   │   └── index.sh
│   ├── index.sh
│   ├── lint.sh
│   ├── node.sh
│   └── nsp.sh
└── update.sh
```

`test/index.sh` invokes `npm run test:lint && npm run test:nsp && npm run test:node && npm run test:browser`.  `test/browser/index.sh` invokes `karma`.  

scripty erroneously executes `test/browser/dev.sh`.  at least.. I *think* that's in error.

**Comment by searls** [comment]
Thanks for this load test.

**Comment by searls** [comment]
Just to clarify:

1. `"test:browser": "scripty"` is in your package.json, confirm?
2. When you run `npm run test:browser`, both `scripts/test/browser/dev.sh` and `scripts/test/browser/index.sh` are being run, right?

What if you rename `index.sh` to `index`? If that fixes it then the `lib/resolve-script/generate-glob` module is probably to blame

**Comment by searls** [comment]
As always I'd appreciate any help you can offer, whether it's a way to reproduce the issue, a test, or a PR. Thanks

**Comment by boneskull** [comment]
> "test:browser": "scripty" is in your package.json, confirm?

yes

> When you run npm run test:browser, both scripts/test/browser/dev.sh and scripts/test/browser/index.sh are being run, right?

yes

> What if you rename index.sh to index? If that fixes it then the lib/resolve-script/generate-glob module is probably to blame

This fixes the issue.

**Comment by boneskull** [comment]
in `generate-glob` you could probably leverage the glob module to avoid both `fs.statSync()` calls, and fix this.

**Comment by boneskull** [comment]
...but that may not be any faster.  sounds silly to generate a globspec using glob

**Comment by boneskull** [comment]
also: ensure `index` is not a directory :smile:

**Comment by boneskull** [comment]
[this line](https://github.com/testdouble/scripty/blob/9e445cc490678d505ba9ac459449549bc8300f2f/lib/resolve-script/generate-glob.js#L10-L10) should probably be

```js
return path.join(expanded, '*')
```

the following

```js
return path.join(expanded, '/') + '*'
```

is not windows-safe, afaik.  the purpose of `join` of course is to insert the proper directory separators for you

**Comment by boneskull** [comment]
```js
return fs.statSync(f + '/index').isFile()
```
isn't windows-safe either, right?  probably want

```js
return fs.statSync(path.join(f, 'index')).isFile()
```

but you may blow that away if you change your strategy anyway

**Comment by boneskull** [comment]
I *think* you can do the index check *before* you do the dir check too.

this might do it:

```fs
// untested
module.exports = function (dir1, dir2) {
  var expanded = path.resolve(dir1, path.join.apply(this, dir2.split(':')))
  // find any index files beneath expanded.  if it's not a directory,
  // this will fail
  var indexGlob = path.join(expanded, 'index*')
  if (glob.sync(indexGlob, {nodir: true}).length) {
    return indexGlob
  }
  // check if expanded is actually a directory
  var dirGlob = path.join(expanded, path.sep)
  if (glob.sync(dirGlob).length) {
    return dirGlob
  }
  // otherwise, default
  return expanded + '*'
}
```

**Comment by searls** [comment]
A couple questions:

1. Are you on Windows?
2. Are you able to reproduce with a test? I tried last night and failed

**Comment by searls** [comment]
Nevermind, was able to reproduce

**Comment by boneskull** [comment]
I'm not on Windows.

**Comment by searls** [comment]
Thanks for your help with this. Fix landed in 1.2.3
